packages/core/src/PhysicsManager.ts

```diff
@@ -0,0 +1,121 @@
+import { Ray, Vector3 } from "@oasis-engine/math";
+import { Layer } from "./Layer";
+import { ColliderFeature } from "./collider";
+import { Engine } from "./Engine";
+import { Collider } from "./collider/Collider";
+
+/**
+ * Structure used to get information back from a raycast or a sweep.
+ */
+export class HitResult {
```

Comment:
Try one class one file in a relatively large framework, you can create a HitResult.ts file.